lib/es/client.rb

```diff
@@ -28,18 +28,32 @@ def initialize(message, error = nil)
     def initialize(es_config, system_logger, crawler_version, crawl_id, &)
       @system_logger = system_logger
       @crawl_id = crawl_id
+
+      # Resolve and store max_retries based on configuration
+      retry_count = es_config.fetch(:retry_on_failure, DEFAULT_MAX_RETRIES) # Default to 3 if not present
+      @max_retries = case retry_count
+                     when true then 3 # Explicitly handle true as 3 retries
+                     when ->(n) { n.is_a?(Integer) && n.positive? } then retry_count # Use positive integer
+                     else 0 # Default to 0 retries for false or invalid values
```

Comment:
Should we invert the condition to set it to 0 for false, but set it to 3 for invalid?